libs/sdk/src/common/notification/notification.ts

```diff
@@ -283,7 +254,18 @@ class NotificationService {
         value: { channelIndex: metaData.channelIndex, data: metaData.data },
       });
     }
-    return notification;
+    // Mark as unread if their SID is greater than the stored SID and add properties for rendering usage
+    Object.defineProperty(notification, 'timestamp', {
+      value: new Date(notification.epoch),
+    });
+
+    const getLatestStoredNotificationID = this.getLatestStoredNotificationID();
```

Comment:
Since `parseNotificationData` is called inside a for loop, it doesn't seem efficient to read local storage inside `getLatestStoredNotificationID` for every iteration. Better call it once and pass it as a param.